pootle/apps/pootle_app/forms.py

```diff
@@ -65,6 +65,11 @@ def __init__(self, *args, **kwargs):
                 self.instance.treestyle == self.instance._detect_treestyle()):
                 self.fields['treestyle'].required = False
 
+    def clean_checkstyle(self):
+        value = self.cleaned_data.get('checkstyle')
+        if value not in PROJECT_CHECKERS.keys():
+            raise forms.ValidationError(_("Invalid checkstyle (%s) " % value))
```

Comment:
I think this validation should be in the model. Otherwise creating the object outside of form doesn't validate.